src/include/units/physical/si/base/thermodynamic_temperature.h

```diff
@@ -95,8 +101,57 @@ namespace unit_constants {
 
+inline constexpr auto zp_kelvin = thermodynamic_temperature_point<kelvin, zero_rep, kelvin_temperature_origin>{};
+
 inline constexpr auto deg_C = thermodynamic_temperature<degree_celsius, one_rep>{};
 
+inline constexpr auto zp_deg_celsius = thermodynamic_temperature_point<degree_celsius, zero_rep, celsius_temperature_origin>{};
+
 }  // namespace unit_constants
 
 }  // namespace units::physical::si
+
+namespace units {
+
+/** \brief Take a quantity of in a standard conventional temperature unit and re-interpret it as a quantity point
+ *
+ * A \c quantity represents "distances" between "points" along a "dimension" scale. However, colloquially, temperature
+ * units are double-used to also represent a "point" directly, to be interpreted as a distance to an implicit reference point.
+ *
+ * This function performs this tasks of identifying and adding that implicit reference point.
+ *
+ * WARNING: The returned value's physical meaning depends on the unit of the input quantity, which has no physical relevance on it's own.
+ * That is, 1 Kelvin and 1 degree Celsius are physically equivalent quantities (as differences between thermodynamic temperatures),
+ * yet the returned temperature points are physically different. This is in stark contrast to the usual behaviour of
+ * this library, where unit conversions. For example, for the following holds:
+ * @code{.cpp}
+ *   auto f = [](auto x) { return 2 * x; }
+ *   assert( quantity_cast<si::degree_celsius>(f(3 * deg_F)) == f( quantity_cast<si::degree_celsius>(3 * deg_F) ) );
+ * @endcode
+ * Here, that is not the case:
+ * @code{.cpp}
+ *   auto f = [](auto x) { return interpret_as_temperature_point(x); }
+ *   assert( quantity_point_cast<si::degree_celsius>(f(3 * deg_F)) != f( quantity_cast<si::degree_celsius>(3 * deg_F) ) );
+ * @endcode
+ *
+ * Therefore, be extra careful when using this function in generic code.
+ *
+ * @param t The quantity to be reinterpreted.
+ * @return A quantity_point using the same representation and value interpreted as a distance to a reference temperature
+ *   which is selected based on the exact unit of the provided quantity.
+ */
+template <QuantityValue Rep>
+inline constexpr auto interpret_as_temperature_point(const units::physical::si::thermodynamic_temperature<units::physical::si::kelvin, Rep> &t) {
+  return units::physical::si::kelvin_temperature_point<units::physical::si::kelvin, Rep>(t);
+}
+template <QuantityValue Rep>
+inline constexpr auto interpret_as_temperature_point(const units::physical::si::thermodynamic_temperature<units::physical::si::millikelvin, Rep> &t) {
+  return units::physical::si::kelvin_temperature_point<units::physical::si::millikelvin, Rep>(t);
+}
+
+template <QuantityValue Rep>
+inline constexpr auto interpret_as_temperature_point(const units::physical::si::thermodynamic_temperature<units::physical::si::degree_celsius, Rep> &t) {
+  return units::physical::si::celsius_temperature_point<units::physical::si::degree_celsius, Rep>(t);
+}
```

Comment:
Do we need this?

Reply:
This is the magic "I guess your origin from the unit you use" function.

Reply:
Hmmm... We could use this in the `quantity_point` CTAD. Have it use the downcasting facility so that temperature units use the correct origin, whereas all other units (and any other program-defined unit that opts in via the downcasting facility) use the default point origin.

Reply:
I find this dangerous if implemented in the constructor. As the comment in the code says, this needs to be used with a lot of care in generic code. So if we put this in the constructor, that means constructors are out from generic code. I doubt that this is desirable. I highlighted a few examples where unexpected things happen in [another comment](https://github.com/mpusz/units/pull/232#discussion_r583087011).